src/audio/drc/drc_math_hifi3.c

```diff
@@ -38,19 +38,12 @@
 #define INV_FUNC_A2_Q25 1126492160 /* Q_CONVERT_FLOAT(33.57208251953125f, 25) */
 #define INV_FUNC_A1_Q25 -713042175 /* Q_CONVERT_FLOAT(-21.25031280517578125f, 25) */
 #define INV_FUNC_A0_Q25 239989712 /* Q_CONVERT_FLOAT(7.152250766754150390625f, 25) */
-
-/*
- * Input depends on precision_x
- * Output range [0.5, 1); regulated to Q2.30
- */
-static inline ae_f32 rexp_fixed(ae_f32 x, int32_t precision_x, int32_t *e)
-{
-	int32_t bit = 31 - AE_NSAZ32_L(x);
-
-	*e = bit - precision_x;
-
-	return AE_SRAA32(x, bit - 30);
-}
+#define SHIFT_IDX_QX30_QY30_QZ30 1 /* drc_get_lshift(30, 30, 30) */
+#define SHIFT_IDX_QX26_QY30_QZ26 1 /* drc_get_lshift(26, 30, 26) */
+#define SHIFT_IDX_QX25_QY30_QZ25 1 /* drc_get_lshift(25, 30, 25) */
+#define SHIFT_IDX_QX29_QY26_QZ26 2 /* drc_get_lshift(29, 26, 26) */
+#define SHIFT_IDX_QX25_QY26_QZ26 6 /* drc_get_lshift(25, 26, 26) */
+#define DRC_TWENTY_Q26 1342177280   /* Q_CONVERT_FLOAT(20, 26) */
```

Comment:
Do you also see cycles increase if you use `#define DRC_TWENTY_Q26 Q_CONVERT_FLOAT(20, 26)`? The purpose of the macro is convenience without producing float code overhead. This was discussed earlier, we could not see emulated or native float code in .lst disassembly from this but @ShriramShastry has seen it to happen with Xplorer GUI toolchain. Do get the same issue?

Reply:
Yes, Seppo. I agree with your comments; Xtensa Xplorer did indicate the greater figure, but on the real device, there is no impact from the experiment disclosed on the MTL platform.

Reply:
I'm still a bit worried about this. If there would be impact in builds for devices from use of Q_CONVERT_FLOAT() we should start to replace them with other macro implementation, if possible or like here directly enter the converted integer. If you see float C code to be generated please file a an issue.

Reply:
Please take note.
I see a few additional additions in the code for `Q_CONVERT_FLOAT() `and `Q_CONVERT_QTOF()` being tried with the HiFi intrinsic instruction set.

[https://github.com/thesofproject/sof/blob/main/src/audio/google/google_rtc_audio_processing.c](url)
```
static inline int16_t convert_float_to_int16(float data)
{
#if XCHAL_HAVE_HIFI3
	const xtfloat ratio = 2 << 15;
	xtfloat x0 = data;
	xtfloat x1;
	int16_t x;

	x1 = XT_MUL_S(x0, ratio);
	x = XT_TRUNC_S(x1, 0);

	return x;
#else /* XCHAL_HAVE_HIFI3 */
	return Q_CONVERT_FLOAT(data, 15);
#endif /* XCHAL_HAVE_HIFI3 */
}

static inline float convert_int16_to_float(int16_t data)
{
#if XCHAL_HAVE_HIFI3
	const xtfloat ratio = 2 << 15;
	xtfloat x0 = data;
	float x;

	x = XT_DIV_S(x0, ratio);

	return x;
#else /* XCHAL_HAVE_HIFI3 */
	return Q_CONVERT_QTOF(data, 15);
#endif /* XCHAL_HAVE_HIFI3 */
}
#endif /* CONFIG_COMP_GOOGLE_RTC_USE_32_BIT_FLOAT_API */
```

Reply:
yeah, I saw cycles increase by xtensa simulator. But I'd like to use the integer directly since then we can reduce the risk of cycles increase on real device(though in our test case it does not) .